src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransfer.cs

```diff
@@ -204,9 +204,12 @@ public int GetSignatureCount(Network network)
             if (scriptSig == null)
                 return 0;
 
+            // Remove the script from the end.
+            scriptSig = new Script(scriptSig.ToOps().SkipLast(1));
```

Comment:
why do you have to do this?
Doesn't `PayToMultiSigTemplate.Instance.ExtractScriptSigParameters` count the signatures properly?

Reply:
`PayToMultiSigTemplate.Instance.ExtractScriptSigParameters` assumes that the op codes are either `OP_0` or an `OP_DATA` with a length corresponding to a valid script signature length. Without the skip the method fails.

Reply:
ah indeed the scriptsig had a redeemscript at the end of it so to count it we need to remove it. Still not the easiest on the eye but that's ok for now.